_docs/geyser/events.md

```diff
@@ -0,0 +1,108 @@
+Extensions do not need to do that - they are automatically registered, so a simple @Subscribe annotation is enough.
+
+**Fabric mod example:**
+```java
+public class ExampleMod implements ModInitializer, EventRegistrar {
+    public static final Logger LOGGER = LoggerFactory.getLogger("modid");
+    
+    @Override 
+    public void onInitialize() {
+        ServerLifecycleEvents.SERVER_STARTING.register((server) -> {
+            GeyserApi.api().eventBus().register(this, ExampleMod.class); // register your mod & this class as a listener
+        });
+        
+        LOGGER.info("Geyser is cool!");
+    }
+    
+    // here an event, we subscribe as usual with the @Subscribe annotation
+    @Subscribe 
+    public void onGeyserPostInitializeEvent(GeyserPostInitializeEvent eventad {
+        LOGGER.info("Geyser started!");
+    }
+}
+```
+<div class="alert alert-info" role="alert">
+	Do note: We cannot directly register the event but in the mod initializer, since the Geyser API would not be loaded yet.
+</div>
+
+Therefore, we register it in the server starting event provided by the Fabric API.
+
+**Paper/Spigot plugin example:**
+
+1. In your plugin.yml, add the following lines:
+```yaml
+  depend: ["Geyser-Spigot"]
+```
+This ensures that your plugin loads after Geyser has, so the Geyser API would be available.
+
+2. In your main class, implement the EventRegistrar interface and register the event bus in the onEnable method:
+```java
+public class ExamplePlugin extends JavaPlugin implements EventRegistrar {
+    
+    @Override
+    public void onEnable(){
+        getLogger().info("Registering Geyser event bus!");
+        GeyserApi.api().eventBus().register(this, this); // register your plugin & this class as a listener
+    }
+
+    // here an event, we subscribe as usual with the @Subscribe annotation
+    @Subscribe
+    public void onGeyserPostInitializeEvent(GeyserPostInitializeEvent event) {
+        getLogger().info("Geyser started!");
+    }
+}
+```
+3. If you want to provide your event with a consumer, rather than annotating it, you can also manually subscribe your method to the event bus:
+```java
+// replace GeyserEvent.class with the event class you want to listen to
+GeyserApi.api().eventBus().subscribe(this, GeyserEvent.class, this::yourMethod);
+```
+
+## Event Priority
+Events can have a priority, which is used to determine the order in which the listeners are called. The default priority is NORMAL.
+The [following priorities](https://github.com/GeyserMC/Events/blob/master/src/main/java/org/geysermc/event/PostOrder.java) are available:
+- ```FIRST``` - called first, before all other listeners
+- ```EARLY``` - called early, before the NORMAL listeners
+- ```NORMAL``` - The normal/default priority
+- ```LATE``` - called late, after the NORMAL listeners
+- ```LAST``` - called last, after all other priority listeners. Generally reserved to monitor the event outcome.
```

Comment:
I'm not sure whether we have to include this, as this is pretty much what the javadoc says

Reply:
Valid point; adjusted